wordpress-mobile/WordPress-Android
# IA reader nav: adding tabs filtering

This PR is one of a set of PR which aims to supersede the #10716 decomposing it in more manageable/testable/reviewable PRs (cc @osullivanchris ).

The current one adds to the FilteredRecyclerView component (currently used in the ReaderPostsListFragment but also in CommentsListFragment and PeopleListFragment) the capability to filter by Tabs in addition to the already present filter by Spinner. In the following a before and after comparison for the Reader.

| Spinner Filtering  | Tab Filtering |
|---|---|
|![image](https://user-images.githubusercontent.com/47797566/68679469-373d8780-0560-11ea-9ad3-92cb1f41cd92.png)|![image](https://user-images.githubusercontent.com/47797566/68679683-a2875980-0560-11ea-84f9-817afbf34f73.png)|

As a result you should get as tabs all the items that were in the spinner and selecting them you should get the relevant list populated as it was selected from the spinner.

Searching and Settings should work as before.

## Notes
- Not adding Release Notes changes since we are still behind feature flagging.
- Current implementation of the tabs with view pager is not using FragmentPagerAdapter/FragmentStatePagerAdapter but a simple PagerAdapter. Fragment sliding will be eventually addressed in future steps.

## To test
Would be good to have both standard and this PR version running (emulator/real device for example) using the same user account so to make behavior comparison easier while doing the following steps.

1. Open the Reader and check you have same items in Spinner/Tabs
2. Select the single items and check the listed content is the same
3. Use the search and check it behaves as expected
4. Access to settings and Subscribe/Unsubscribe to Tags and Sites and check that when you come back to the reader the items are added/removed correctly
5. Select a Tag/Site then close the app and open it again. The reader should be presented with the last selected Tag/Site.
6. Smoke test the reader in general (open posts from the reader, follow/unfollow tags/sites etc...)
7. Smoke test the other parts in which the FilteredRecyclerView is used (namely My Site > Comments and My Site > People)

PR submission checklist:
